openstudiocore/src/model/FileOperations.cpp

```diff
@@ -79,242 +79,69 @@
-      if (!test){
-        LOG_FREE(Error, "removeDir", "Failed to remove directory '" << toString(dirName) << "'");
-        result = false;
-
-        // DLM: do we really want to give up here?
-        //return result;
-      }
-    }
-
-    return result;
-  }
-
-  bool synchDirStructures(const QString &srcPath, const QString &dstPath)
-  {
-    bool result = true;
-    bool test = true;
-
-    QDir srcDir(srcPath);
-    QDir dstDir(dstPath);
-
-    if (srcDir.canonicalPath() == dstDir.canonicalPath()){
-      LOG_FREE(Warn, "synchDirStructures", "Cannot synch destination '" << toString(dstPath) << "' with source '" << toString(srcPath) << "' because they resolve to the same location");
-      return true; // already synched
+    if (!openstudio::filesystem::exists(sourceDir) || !openstudio::filesystem::is_directory(sourceDir))
+    {
+      throw std::runtime_error("Source directory " + sourceDir.string() + " does not exist or is not a directory");
     }
 
-    LOG_FREE(Info, "synchDirStructures", "Synching destination '" << toString(dstPath) << "' with source '" << toString(srcPath) << "'");
-
-    // remove all files in dst as well as any directories in dst that are not in src
-    for (const QFileInfo &dstItemInfo : dstDir.entryInfoList(QDir::Dirs | QDir::Files | QDir::NoDotAndDotDot))
+    if (openstudio::filesystem::exists(destinationDir))
     {
-      QString srcItemPath = srcPath + "/" + dstItemInfo.fileName();
-      QString dstItemPath = dstPath + "/" + dstItemInfo.fileName();
-
-      QFileInfo srcItemInfo = QFileInfo(srcItemPath);
-
-      if (dstItemInfo.isDir()){
-
-        if (srcItemInfo.exists() && srcItemInfo.isDir()){
-
-          // directory also exists in source
-          test = synchDirStructures(srcItemPath, dstItemPath);
-          if (!test){
-            LOG_FREE(Error, "synchDirStructures", "Failed to synch destination '" << toString(dstItemPath) << "' with source '" << toString(srcItemPath) << "'");
-            result = false;
-
-            // DLM: do we really want to give up here?
-            //return result;
-          }
-
-        } else{
-
-          // directory does not exist in source
-          test = removeDir(dstItemPath);
-          if (!test){
-            LOG_FREE(Error, "synchDirStructures", "Failed to remove directory '" << toString(dstItemPath) << "'");
-            result = false;
-
-            // DLM: do we really want to give up here?
-            //return result;
-          }
-        }
-
-      } else if (dstItemInfo.isFile()){
-        test = QFile::remove(dstItemInfo.absoluteFilePath());
-        if (!test){
-          LOG_FREE(Error, "synchDirStructures", "Failed to remove file '" << toString(dstItemInfo.absoluteFilePath()) << "'");
-          result = false;
-
-          // DLM: do we really want to give up here?
-          //return result;
-        }
-      }
+      throw std::runtime_error("Destination directory " + destinationDir.string() + " already exists");
```

Comment:
@lefticus this is obviously much simpler than what was there.  It will fail if the destination already exists, so we'll need to ensure that we are deleting the destination before we call this.  Can you do a text search to see where these are being called and check that we are doing that?  It is also throwing when previously would return false.  Can we replace these throws with logging and error and returning false?

Reply:
Maybe we should delete the destination in this call?

Reply:
I think this depends on if we want to stick with this more simplified behavior or recreate the older more complex sync behavior? It would seem that a viable option is to delete the existing dir and copy this one in to it? This should negate the need for the sync? (but I admittedly don't understand the need for the complexity of the old version)

Reply:
@macumber I just confirmed that the previous code would "synchronize" by simply deleting any files and folders that were not in the destination. It would seem much simpler to just delete the existing folder and copy over the new one.

Reply:
Cool that seems ok to me.  I think the things I would ask for are:

1. Trap for case where src and dest are the same (have to get the lexically correct path to compare)
1. Delete the destination if it exists (don't require this outside the call)
1. Use log error and return false instead of throw

Reply:
Updated in 5d54887cf1990b6230771ac05e76f56ebfd60286